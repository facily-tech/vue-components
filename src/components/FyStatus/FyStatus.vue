<template>
  <fy-card-form class="fy-status">
    <template #title>
      {{ title }}
    </template>
    <v-card-text>
      <v-row>
        <v-col md="10" cols="12">
          <div class="subtitle-1 font-weight-black">
            {{ value ? active : inactive }}
          </div>
          <div v-if="description">
            {{ description }}
          </div>
        </v-col>
        <v-col md="2" cols="12" class="d-flex justify-end">
          <v-switch v-model="model" inset data-test="select-status" :disabled="disabled" />
        </v-col>
      </v-row>
    </v-card-text>
  </fy-card-form>
</template>

<script lang="ts">
import Vue from 'vue';
import FyCardForm from '../FyCard/FyCardForm.vue';

export default Vue.extend({
  components: { FyCardForm },
  name: 'FyStatus',

  props: {
    value: {
      type: Boolean,
      required: true,
    },

    title: {
      type: String,
      default: 'Status',
    },

    description: {
      type: String,
    },

    active: {
      type: String,
      default: 'Ativo',
    },

    inactive: {
      type: String,
      default: 'Inativo',
    },

    disabled: {
      type: Boolean,
      default: false,
    },
  },

  computed: {
    model: {
      get(): boolean {
        return this.value;
      },

      set(value: boolean) {
        this.$emit('input', value);
      },
    },
  },
});
</script>
