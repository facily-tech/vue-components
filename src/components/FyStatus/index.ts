export { default as FyStatus } from './FyStatus.vue';
