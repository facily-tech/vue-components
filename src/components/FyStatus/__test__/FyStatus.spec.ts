import Vue from 'vue';
import Router from 'vue-router';
import FyStatus from '../FyStatus.vue';
import { createLocalVue, mount, Wrapper } from '@vue/test-utils';
import Vuetify from 'vuetify';

Vue.use(Vuetify);

describe('FyStatus', () => {
  // eslint-disable-next-line
  let router: Router;
  let localVue: typeof Vue;

  beforeEach(() => {
    router = new Router();
    localVue = createLocalVue();
    localVue.use(Router);
  });

  it('should render component and match snapshot', () => {
    const wrapper = mount(FyStatus, {
      localVue,
      vuetify: new Vuetify(),
      router,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('when switch is clicked', () => {
    const wrapper: Wrapper<any> = mount(FyStatus, {
      localVue,
      vuetify: new Vuetify(),
      router,
    });

    expect(wrapper.emitted('input')).toBeFalsy();

    wrapper.find('[data-test="select-status"]').trigger('click');

    expect(wrapper.emitted('input')).toBeTruthy();
  });
});
