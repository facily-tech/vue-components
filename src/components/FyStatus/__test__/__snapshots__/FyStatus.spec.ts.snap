// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyCard should render component and match snapshot 1`] = `
<div class="fy-card-form pa-0 pa-md-4 fy-status v-card v-sheet theme--light elevation-0">
  <div class="v-card__title fy-card-title text-h6 font-weight-medium justify-space-between">
    Status
  </div>
  <!---->
  <div class="v-card__text">
    <div class="row">
      <div class="col-md-10 col-12">
        <div class="subtitle-1 font-weight-black">
          Inativo
        </div>
        <!---->
      </div>
      <div class="d-flex justify-end col-md-2 col-12">
        <div class="v-input theme--light v-input--selection-controls v-input--switch v-input--switch--inset">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-input--selection-controls__input"><input aria-checked="false" id="input-5" role="switch" type="checkbox" aria-disabled="false" data-test="select-status" value="">
                <div class="v-input--selection-controls__ripple"></div>
                <div class="v-input--switch__track theme--light"></div>
                <div class="v-input--switch__thumb theme--light">
                  <transition-stub name="fab-transition" mode="out-in"></transition-stub>
                </div>
              </div>
            </div>
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`FyStatus should render component and match snapshot 1`] = `
<div class="fy-card-form pa-0 pa-md-4 fy-status v-card v-sheet theme--light elevation-0">
  <div class="v-card__title fy-card-title text-h6 font-weight-medium justify-space-between">
    Status
  </div>
  <!---->
  <div class="v-card__text">
    <div class="row">
      <div class="col-md-10 col-12">
        <div class="subtitle-1 font-weight-black">
          Inativo
        </div>
        <!---->
      </div>
      <div class="d-flex justify-end col-md-2 col-12">
        <div class="v-input theme--light v-input--selection-controls v-input--switch v-input--switch--inset">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-input--selection-controls__input"><input aria-checked="false" id="input-5" role="switch" type="checkbox" aria-disabled="false" data-test="select-status" value="">
                <div class="v-input--selection-controls__ripple"></div>
                <div class="v-input--switch__track theme--light"></div>
                <div class="v-input--switch__thumb theme--light">
                  <transition-stub name="fab-transition" mode="out-in"></transition-stub>
                </div>
              </div>
            </div>
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
